SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorLayoutDetails.cpp

```diff
@@ -124,65 +126,20 @@ void FSpatialGDKEditorLayoutDetails::CustomizeDetails(IDetailLayoutBuilder& Deta
+	FString DevAuthToken;
+	FString ErrorMessage;
+	if (!SpatialCommandUtils::GenerateDevAuthToken(GetDefault<USpatialGDKSettings>()->IsRunningInChina(), DevAuthToken, ErrorMessage))
 	{
-		UE_LOG(LogSpatialGDKEditorLayoutDetails, Error, TEXT("Unable to generate a development authentication token. Result: %s"), *CreateDevAuthTokenResult);
-		FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(FString::Printf(TEXT("Unable to generate a development authentication token. Result: %s"), *CreateDevAuthTokenResult)));
-		return FReply::Unhandled();
-	};
-
-	FString AuthResult;
-	FString DevAuthTokenResult;
-	bool bFoundNewline = CreateDevAuthTokenResult.TrimEnd().Split(TEXT("\n"), &AuthResult, &DevAuthTokenResult, ESearchCase::IgnoreCase, ESearchDir::FromEnd);
-	if (!bFoundNewline || DevAuthTokenResult.IsEmpty())
-	{
-		// This is necessary because depending on whether you are already authenticated against spatial, it will either return two json structs or one.
-		DevAuthTokenResult = CreateDevAuthTokenResult;
-	}
-
-	TSharedRef<TJsonReader<TCHAR>> JsonReader = TJsonReaderFactory<TCHAR>::Create(DevAuthTokenResult);
-	TSharedPtr<FJsonObject> JsonRootObject;
-	if (!(FJsonSerializer::Deserialize(JsonReader, JsonRootObject) && JsonRootObject.IsValid()))
-	{
-		UE_LOG(LogSpatialGDKEditorLayoutDetails, Error, TEXT("Unable to parse the received development authentication token. Result: %s"), *DevAuthTokenResult);
-		FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(FString::Printf(TEXT("Unable to parse the received development authentication token. Result: %s"), *DevAuthTokenResult)));
-		return FReply::Unhandled();
-	}
-
-	// We need a pointer to a shared pointer due to how the JSON API works.
-	const TSharedPtr<FJsonObject>* JsonDataObject;
-	if (!(JsonRootObject->TryGetObjectField("json_data", JsonDataObject)))
-	{
-		UE_LOG(LogSpatialGDKEditorLayoutDetails, Error, TEXT("Unable to parse the received json data. Result: %s"), *DevAuthTokenResult);
-		FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(FString::Printf(TEXT("Unable to parse the received json data. Result: %s"), *DevAuthTokenResult)));
-		return FReply::Unhandled();
-	}
-
-	FString TokenSecret;
-	if (!(*JsonDataObject)->TryGetStringField("token_secret", TokenSecret))
-	{
-		UE_LOG(LogSpatialGDKEditorLayoutDetails, Error, TEXT("Unable to parse the token_secret field inside the received json data. Result: %s"), *DevAuthTokenResult);
-		FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(FString::Printf(TEXT("Unable to parse the token_secret field inside the received json data. Result: %s"), *DevAuthTokenResult)));
+		FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(ErrorMessage));
 		return FReply::Unhandled();
 	}
 
 	if (USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>())
 	{
-		SpatialGDKEditorSettings->DevelopmentAuthenticationToken = TokenSecret;
+		SpatialGDKEditorSettings->DevelopmentAuthenticationToken = DevAuthToken;
```

Comment:
Yeah this would be nice to wrap up on a function inside SpatialGDKEditorSettings: SetDevAuthToken()